### PR TITLE
Proper BN momentum parameter and train mode setting in BN adaptation

### PR DESCRIPTION
**Pull request by vanyalzr**
Fix BN adaptation due to #257

### CODE REVIEW

File: nncf/initialization.py [hunk 1]
```diff
@@ -210,6 +224,8 @@ def set_bn_momentum(module, momentum_value):
             args_kwargs_tuple = data_loader.get_inputs(loaded_item)
             self._infer_batch(args_kwargs_tuple, device)
 
+        self.model.eval()
```
Comment (ljaljushkin) [inline nncf/initialization.py:227]: Not sure we should always set the model to eval mode. It can be not obvious for the users of this init functionality.

I'd add a context manager or decorator for secure safe switching modes.

The idea is always to keep the original state of the model with activating the chosen mode inside a function's body. 
```
def _run_model_inference
    # doesn't matter what is a current training mode of the model, it would be returned to the original state outside of the context manager
    with training_mode_switcher(model, is_training=True) as model:
         # work with the model in a training mode
         model.infer()
    # outside of context manager model is in the original sate
    model.infer()
```

That would be helpful in multiple places in NNCF.

**Comment by vshampor** [comment]
Jenkins please retry a build

File: nncf/initialization.py (hunk 1 above)
Comment (vshampor) [reply nncf/initialization.py:227]: @vanyalzr are you going to address this?

Comment (vanyalzr) [reply nncf/initialization.py:227]: Sure, I agree with the comment, will address

**Review by ljaljushkin — APPROVED** [review]
